new-components/enrichers/custom-annotation/internal/annotation/annotation_test.go

```diff
@@ -0,0 +1,253 @@
+					Uid:           "2",
+				},
+				Message: ptr("lots of hacky code"),
+				Resource: &ocsf.ResourceDetails{
+					Uid: ptr(
+						strings.Join([]string{
+							"/internal/sketchy/sketch.go",
+							"10",
+							"1",
+						},
+							":",
+						),
+					),
+					Data: &structpb.Value{
+						Kind: &structpb.Value_StringValue{
+							StringValue: "2",
+						},
+					},
+				},
+				RawData:    ptr(`{"issues" : [{"id": 2}]}`),
+				Severity:   ptr("HIGH"),
+				SeverityId: ocsf.VulnerabilityFinding_SEVERITY_ID_HIGH,
+				StartTime:  &now,
+				Status:     ptr("opened"),
+				Time:       now,
+				TypeUid: int64(
+					ocsf.VulnerabilityFinding_CLASS_UID_VULNERABILITY_FINDING.Number()*
+						100 +
+						ocsf.VulnerabilityFinding_ACTIVITY_ID_CREATE.Number(),
+				),
+				Vulnerabilities: []*ocsf.Vulnerability{
+					{
+						Cwe: &ocsf.Cwe{
+							Uid:    "2",
+							SrcUrl: ptr("https://issues.com/2"),
+						},
+					},
+				},
+			},
+			{
+				ActivityId:   ocsf.VulnerabilityFinding_ACTIVITY_ID_CREATE,
+				CategoryUid:  ocsf.VulnerabilityFinding_CATEGORY_UID_FINDINGS,
+				ClassUid:     ocsf.VulnerabilityFinding_CLASS_UID_VULNERABILITY_FINDING,
+				Confidence:   ptr("LOW"),
+				ConfidenceId: ptr(ocsf.VulnerabilityFinding_CONFIDENCE_ID_LOW),
+				Count:        ptr(int32(3)),
+				FindingInfo: &ocsf.FindingInfo{
+					CreatedTime: &now,
+					DataSources: []string{
+						"/internal/sketchy/hacks.go",
+					},
+					Desc:          ptr("stop writing hacks"),
+					FirstSeenTime: &now,
+					LastSeenTime:  &now,
+					ModifiedTime:  &now,
+					ProductUid:    ptr("gosec"),
+					Title:         "You have lots of hacks",
+					Uid:           "3",
+				},
+				Message: ptr("lots of hacks"),
+				Resource: &ocsf.ResourceDetails{
+					Uid: ptr(
+						strings.Join([]string{
+							"/internal/sketchy/hacks.go",
+							"123",
+							"13",
+						},
+							":",
+						),
+					),
+					Data: &structpb.Value{
+						Kind: &structpb.Value_StringValue{
+							StringValue: "3",
+						},
+					},
+				},
+				RawData:    ptr(`{"issues" : [{"id": 3}]}`),
+				Severity:   ptr("HIGH"),
+				SeverityId: ocsf.VulnerabilityFinding_SEVERITY_ID_HIGH,
+				StartTime:  &now,
+				Status:     ptr("opened"),
+				Time:       now,
+				TypeUid: int64(
+					ocsf.VulnerabilityFinding_CLASS_UID_VULNERABILITY_FINDING.Number()*
+						100 +
+						ocsf.VulnerabilityFinding_ACTIVITY_ID_CREATE.Number(),
+				),
+				Vulnerabilities: []*ocsf.Vulnerability{
+					{
+						Cwe: &ocsf.Cwe{
+							Uid:    "3",
+							SrcUrl: ptr("https://issues.com/3"),
+						},
+					},
+				},
+			},
+		}
+	)
+
+	annotationValuesAsJson, err := json.Marshal(annotationValues)
+	require.NoError(t, err)
+
+	defer cancel()
+
+	t.Run("it should enrich the issues successfully", func(t *testing.T) {
+		annotator, err := annotation.NewCustomAnnotator(conf)
+		require.NoError(t, err)
+
+		enrichedVulns, err := annotator.Annotate(ctx, vulns)
```

Comment:
I think you can use the `.ElementsMatch` function to check the lists more easily over here